servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -351,10 +359,16 @@ public BlockMeta moveBlockMeta(BlockMeta blockMeta, TempBlockMeta tempBlockMeta)
       throws NotFoundException, OutOfSpaceException, AlreadyExistsException {
     StorageDir srcDir = blockMeta.getParentDir();
     StorageDir dstDir = tempBlockMeta.getParentDir();
+
+    dstDir.removeTempBlockMeta(tempBlockMeta);
+
+    if (srcDir.equals(dstDir)) {
+      return blockMeta;
+    }
+
```

Comment:
I think its okay to leave out the check then, and we are not saving much anyway, its better if the code is simpler.